lib/ansible/modules/cloud/online/online_manage_c14.py

```diff
@@ -0,0 +1,496 @@
+            
+        self.name = "storage/c14/safe/{0}/archive/{1}".format(safe,archive_uuid)
+        delete_data = dict()
+        result = self.delete('/%s' % self.name, delete_data)
+        
+        self.module.exit_json(result='The archive {0} is scheduled to be removed'.format(archive_uuid))
+
+    def get_archive_key(self, archive_uuid):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)
+        self.name = "storage/c14/safe/{0}/archive/{1}/key".format(safe,archive_uuid)
+        key = self.get('/%s' % self.name)
+        self.module.exit_json(result=key.json)
+        
+    def set_archive_key(self, archive_uuid, archive_key):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)
+        self.name = "storage/c14/safe/{0}/archive/{1}/key".format(safe,archive_uuid)
+        post_data = dict(
+            key = archive_key
+        )
+        result = self.post('/%s' % self.name, post_data)
+        if result.json is None:
+            self.module.exit_json(result="The archive key has been set")
+        else:        
+            self.module.exit_json(result=result.json)        
+
+    def delete_archive_key(self, archive_uuid, archive_key):
+        safe = self.find_safe_uuid_from_archive(archive_uuid)
+        self.name = "storage/c14/safe/{0}/archive/{1}/key".format(safe,archive_uuid)
+        delete_data = dict()
+        result = self.delete('/%s' % self.name, delete_data)
+        if result.json is None:
+            self.module.exit_json(result="The archive key has been deleted")
+        else:
+            self.module.exit_json(result=result.json)          
+        
+def main():
+    argument_spec = online_argument_spec()
+    argument_spec.update(
+        dict(
+            operation=dict(
+                required=True,
+                choices=['get_archives','get_safes','get_bucket','get_files','get_archive_key','set_archive_key','delete_archive_key','create_archive','upload_files','download_files','freeze_archive','unfreeze_archive','remove_archive','remove_files']
+            ),
+            archive_uuid=dict(),
+            archive_key=dict(),
+            archive_details=dict(
+                type = dict
+            ),
+            paths=dict(
+                type = list,
+                aliases=['path']
+            ),
+            download_path=dict(
+                default='/tmp'
+            ),
+            wait=dict(
+                default=False
+            )
+        )
+    )
+    
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    operation = module.params['operation']
+    archive_uuid = module.params['archive_uuid']
+    paths =  module.params['paths']
+    download_path =  module.params['download_path']
+    archive_details =  module.params['archive_details']    
+    archive_key =  module.params['archive_key']
+    wait = module.params['wait']
+    
+    try:
+        if operation == 'get_archives':
```

Comment:
Instead of having very long `elif` loop. Could you have a dict that got the name of an operation as a key and the function to apply as a value. That way we avoid doing a good deal of useless string comparisons.